tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -400,11 +328,7 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 				}
 			}
 
-			test.emit("benchmark end", benchmarkStats);
-			await testObject.after?.();
```

Comment:
This should still be called before `return benchmarkStats`, right?

Reply:
I think so, but we have tests for that which are passing. I'll investigate.

Reply:
Oh, I forgot we have to `pnpm run perf` to test this code path, and that was broken until recently. With that fix merged into this branch, it doesn't show any error: I had to add `cross-env FLUID_TEST_VERBOSE=1` to errors in tests to show (so I did that).

I also added another error case where we print errors in hooks, and refactored benchmarkMemory so its correctness and measurement cases share mode code for improved coverage.

I have manually confirmed that this bug is now detected and properly logged by the correctness tests, "perf" and when running with --parentProcess.